libvmi/driver/xen/xen.c

```diff
@@ -165,7 +166,183 @@ xen_put_memory(
     return VMI_SUCCESS;
 }
 
+/* Helper function. Read /local/domain/<domID>/device/vbd/state property.
+ * If it equals "1" return true.
+ */
+#ifndef HAVE_LIBXENSTORE
+static bool
+xen_get_vbd_state(
+    vmi_instance_t vmi,
+    const char* device_id)
+{
+    return false;
+}
+#else
+static bool
+xen_get_vbd_state(
+    vmi_instance_t vmi,
+    const char* device_id)
+{
+    unsigned int len;
+    bool result = false;
+    xen_instance_t *xen = xen_get_instance(vmi);
+    xs_transaction_t xth = XBT_NULL;
+
+    struct xs_handle *xsh = xen->libxsw.xs_open(0);
```

Comment:
Any particular reason you open a new xs handle and not just use the one already open?

Reply:
`xen_get_vbd_state` function is called at the beginning of `xen_read_disk` and there is no opened handle at this time, or am i missing something?

Reply:
The xen instance has an active xenstore handle after `xen_init`.